Source/Core/LMCHFSSResponseFileHandler.cc

```diff
@@ -108,14 +114,24 @@ namespace locust
             fTFComplex[i][0]=tfArray.at(i).real();
             fTFComplex[i][1]=tfArray.at(i).imag();
         }
-        fComplexFFT.SetupIFFT(fTFNBins,fInitialTFIndex,fTFBinWidth);
-	fFIRNBins=fTFNBins+2*fComplexFFT.GetShiftNBins();
+	double TFBinWidth = 2./9.*fInitialTFIndex/(1.0*fTFNBins); //Calculates TFBinWidth assuming frequency range defined as .9 to 1.1 the center of the frequency bins.
+	if(GeneratedTF){ 
+		fComplexFFT.SetupIFFT(fTFNBins,fInitialTFIndex,TFBinWidth);//Uses binwidth as calculated in the previous line based on internally generated TF
+	}
+	else{ 
+		fComplexFFT.SetupIFFT(fTFNBins,fInitialTFIndex,fTFBinWidth); //Uses degenerately defined fTFNBins AND fTFBinWidth if using external TF function.
+	}
+	fFIRNBins=fTFNBins+2*2*fComplexFFT.GetShiftNBins();
```

Comment:
Could this also be put inside of the if-statements above? Where the analytically generated FIR Bins is longer, but the numerical FIR is shorter? I think this is a bit too long for the HFSS TFs.

Reply:
While I definitely have a typo of an extra factor of 2 on the `GetShiftNBins()` term, are you suggesting that this term should be defined differently for each of the two cases (analytically generated TF vs external TF)? As of now, they both use the same two parameters from the config files. As of now only the bin width is defined differently (this is done to preserve some backwards compatibility of the code), but that isn't used to calculate `fFIRNBins`.

Reply:
Sorry - I thought the typo was intentional and necessary for the generated TF. If you just get rid of the extra 2, then this is resolved.

Reply:
Why not just assign `2./9.*fInitialTFIndex/(1.0*fTFNBins)` to `fTFBinWidth` directly. So something like for line 117:
```if(GeneratedTF)
{
     double fTFBinWidth = 2./9.*fInitialTFIndex/(1.0*fTFNBins);
}
```
And then you don't need the if-else condition here and you can just do
``` 
fComplexFFT.SetupIFFT(fTFNBins,fInitialTFIndex,fTFBinWidth);
```

Reply:
In fact why not do define `fTFBinWidth` in `GenerateAnalyticTFtoFIR` right after you define `fTFNBins` ? You won't need these additional if loops at all.

Reply:
The tricky part here is that I'm attempting to keep this framework flexible so that it still works if being run the original way for when users don't want to use the equivalent circuit model. The original way defines the range of the frequencies explicitly in the config file via the number of bins as well as the bin width (where the initial frequency is taken directly from the external TF file). With the loops, this allows the FFT to either be setup using the bin width as calculated from the internally generated TF function or using the parameters defined in the earlier method. There certainly may be a cleaner way to do this, but I wanted to try to leave the original method as untouched as possible.